### ISSUE TITLE
[feature request] Parse C++20 modules

### ISSUE THREAD

**Issue by alfaix** [tree-sitter/tree-sitter-cpp#174]
Hi!

I'm experimenting with C++20 modules in neovim and tree-sitter can't parse modules.
Details regarding the syntax and my implementation can be found in [this PR](https://github.com/tree-sitter/tree-sitter-cpp/pull/173).

**Comment by cdacamar** [comment]
I would love to see this.  Currently translation units containing C++ modules keywords are hard to follow as highlighting quality quickly takes a nose-dive particularly after `export` is attached to various declarations.
